WordPress/JetpackStatsWidgets/Tracks/WidgetAnalytics.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import WidgetKit
+
+@objcMembers class WidgetAnalytics: NSObject {
+    static func trackLoadedWidgetsOnApplicationOpened() {
+        guard AppConfiguration.isJetpack else { return }
+
+        WidgetCenter.shared.getCurrentConfigurations { result in
+            let properties = self.properties(from: result)
+            WPAnalytics.track(.widgetsLoadedOnApplicationOpened, properties: properties)
+        }
+    }
+
+    private static func properties(from widgetInfo: Result<[WidgetInfo], Error>) -> [String: String] {
+        guard let installedWidgets = try? widgetInfo.get() else {
+            return ["widgets": ""]
+        }
+
+        let widgetAnalyticNames: [String] = installedWidgets.map { widgetInfo in
+            guard let eventKind = AppConfiguration.Widget.Stats.Kind(rawValue: widgetInfo.kind) else {
+                DDLogWarn("⚠️ Make sure the widget: \(widgetInfo.kind), has the correct kind.")
+                return "\(widgetInfo.kind)_\(widgetInfo.family)"
+            }
+            return "\(Events.eventPrefix(for: eventKind))_\(widgetInfo.family)"
+        }
+
+        return ["widgets": widgetAnalyticNames.joined(separator: ",")]
+    }
```

Comment:
The comma-separated list that this produces could be tricky to analyze or filter since all the widget names are together in a single property:

```
🔵 Tracked: widgets_loaded_on_application_opened <widgets: homeThisWeekWidget_systemMedium,homeAllTimeWidget_systemMedium,lockScreenTodayViewsVisitorsWidget_accessoryRectangular>
```


 Should multiple properties be used instead? Something like this:
```
🔵 Tracked: widgets_loaded_on_application_opened <today_home_extension_widget_systemmedium: true, today_home_extension_widget_systemsmall: true>
```

The idea is that any unique widget is included as a property of the event. Any widgets not present aren't mapped to a property.

I tested this code to do this:

```
            return [:]
        }

        let widgetAnalyticNames: [String] = installedWidgets.map { widgetInfo in
            guard let eventKind = AppConfiguration.Widget.Stats.Kind(rawValue: widgetInfo.kind) else {
                DDLogWarn("⚠️ Make sure the widget: \(widgetInfo.kind), has the correct kind.")
                return "\(widgetInfo.kind)_\(widgetInfo.family)"
            }
            return "\(Events.eventPrefix(for: eventKind).rawValue)_\(widgetInfo.family.description.lowercased())"
        }

        let dict = Dictionary(uniqueKeysWithValues: widgetAnalyticNames.map { name in
            return (name, "true")
        })

        return dict
    }
```

Here I used "true" as a value for lack of a better alternative (there's no "false" value because widgets that aren't added won't be included in the dictionary). Tracks also wanted the event property to use lowercase.

Reply:
Yes, you're right. I used the example of Woo, and I thought that the tracks system interprets commas more cleverly, allowing us to see the total usage of all the widgets separately under one property. However, that is not the case. 

Your proposal definitely makes sense in this instance. I also updated property keys so they would all start with `widget_` prefix, for easier visibility within tracks.